examples/a_treeple/treeple_tutorial_CMI.py

```diff
@@ -0,0 +1,149 @@
+# *X* conditioned on *Z*. It can be calculated by the difference between
+# the joint MI (``I([X, Z]; Y)``) and the MI on Z (``I(Y; Z)``):
+#
+# .. math:: I(X; Y | Z) = I([X, Z]; Y) - I(Y; Z)
+#
+# With a multiview binary class simulation as an example, this tutorial
+# will show how to use ``treeple`` to calculate the statistic with
+# multiview data.
+
+# %%
+# Create a simulation with two gaussians
+# --------------------------------------
+
+
+# create a binary class simulation with two gaussians
+# 500 samples for each class, class zero is standard
+# gaussian, and class one has a mean at one for Z
+Z, y = make_trunk_classification(
+    n_samples=1000,
+    n_dim=1,
+    mu_0=0,
+    mu_1=1,
+    n_informative=1,
+    seed=1,
+)
+
+
+X, y = make_trunk_classification(
+    n_samples=1000,
+    n_dim=1,
+    mu_0=0,
+    mu_1=2,
+    n_informative=1,
+    seed=1,
+)
+
+
+# scatter plot the samples for Z
+plt.hist(Z[500:], bins=15, alpha=0.6, color="blue", label="negative")
+plt.hist(Z[:500], bins=15, alpha=0.6, color="red", label="positive")
+plt.legend()
+plt.show()
+
+
+# scatter plot the samples for X
+plt.hist(X[500:], bins=15, alpha=0.6, color="blue", label="negative")
+plt.hist(X[:500], bins=15, alpha=0.6, color="red", label="positive")
+plt.legend()
+plt.show()
+
+
+# %%
+# Fit the model with X and Z
+# --------------------------
+
+
+# initialize the forest with 100 trees
+est = HonestForestClassifier(
+    n_estimators=100,
+    max_samples=1.6,
+    max_features=0.3,
+    bootstrap=True,
+    stratify=True,
+    tree_estimator=MultiViewDecisionTreeClassifier(),
+    random_state=1,
+)
+
+# fit the model and obtain the tree posteriors
+_, observe_proba = build_hyppo_oob_forest(est, np.hstack((X, Z)), y)
+
+# generate forest posteriors for the two classes
+observe_proba = np.nanmean(observe_proba, axis=0)
+
+
+# scatter plot the posterior probabilities for class one
+plt.hist(observe_proba[:500][:, 1], bins=30, alpha=0.6, color="blue", label="negative")
+plt.hist(observe_proba[500:][:, 1], bins=30, alpha=0.6, color="red", label="positive")
+plt.legend()
+plt.show()
+
+# %%
+# Fit the model with Z only
+# -------------------------
+
+
+# initialize the forest with 100 trees
+est = HonestForestClassifier(
+    n_estimators=100,
+    max_samples=1.6,
+    max_features=0.3,
+    bootstrap=True,
+    stratify=True,
+    random_state=1,
+)
+
+# fit the model and obtain the tree posteriors
+_, single_proba = build_hyppo_oob_forest(est, Z, y)
+
+# generate forest posteriors for the two classes
+single_proba = np.nanmean(single_proba, axis=0)
+
+
+# scatter plot the posterior probabilities for class one
+plt.hist(single_proba[:500][:, 1], bins=30, alpha=0.6, color="blue", label="negative")
+plt.hist(single_proba[500:][:, 1], bins=30, alpha=0.6, color="red", label="positive")
+plt.legend()
+plt.show()
+
+
+# %%
+# Calculate the statistic
+# -----------------------
+
+
+def Calculate_MI(y_true, y_pred_proba):
```

Comment:
Is it worth making the existing function we have public so you don't need to redefine one here?

Reply:
That would be best.